types/hippie/index.d.ts

```diff
@@ -0,0 +1,62 @@
+// Type definitions for hippie 0.5
+// Project: https://github.com/vesln/hippie
+// Definitions by: Nico Muschert <https://github.com/nicomuschert>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// TypeScript Version: 2.2
+/// <reference types="node" />
+
+import http = require('http');
+
+interface Err {
+  AssertionError: string;
+  message: string;
+  showDiff: boolean;
+  actual: boolean;
+  expected: boolean;
+}
+
+interface EndCb {
+  (err: undefined|Err|Error, res: http.IncomingMessage, body?: object|string): void;
+}
+
+interface Opts {
+  [key: string]: string|{};
+}
+
+declare function hippie(app?: any): typeof hippie;
+declare namespace hippie {
+    function end(end?: EndCb): undefined|Promise<any>;
+    function del(url: string, fn?: EndCb): typeof hippie;
+    function get(url: string, fn?: EndCb): typeof hippie;
+    function head(url: string, fn?: EndCb): typeof hippie;
+    function patch(url: string, fn?: EndCb): typeof hippie;
+    function post(url: string, fn?: EndCb): typeof hippie;
+    function put(url: string, fn?: EndCb): typeof hippie;
+    function method(method: string): typeof hippie;
+    function url(url: string): typeof hippie;
+    function header(key: string, val: string): typeof hippie;
+    function base(url: string): typeof hippie;
+    function app(app: (req: http.ClientRequest, res: http.ServerResponse) => {}): typeof hippie;
+    function auth(user: string, pass: string): typeof hippie;
+    function opts(fn: (opts: Opts) => void): typeof hippie;
+    function form(): typeof hippie;
+    function json(): typeof hippie;
+    function qs(obj: {}): string;
+    function send(data: any): typeof hippie;
+    function expect(fn: (result: http.IncomingMessage, body: string|{}, next: (err?: Err|Error) => void) => void): typeof hippie;
```

Comment:
As discussed [here](http://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html), it's not recommended to have optional parameters in callbacks.

Reply:
Hi Pranav, thank you for review.

Hippie is not my baby. I would like to express how it is expected to work with it based on vesln's (the author) examples. But i think i can change the definition without having side effects in this way:

`err: null|Err|Error`

Would you recommend it with a that null-type?

Best,
Nico

Reply:
You should be able to just make it non-optional. The user can still pass in a callback that takes no arguments so there shouldn't be any change in usage (but has the benefits described in the link above).

```suggestion
    function expect(fn: (result: http.IncomingMessage, body: string|{}, next: (err: Err|Error) => void) => void): typeof hippie;
```